File: benchmark/comparison/results/blind_py/17.py
def zip_lists(a, b):
    result = []
    for i in range(min(len(a), len(b))):
        result.append((a[i], b[i]))
    return result

def fmt(pairs):
    items = [f"Pair({a}, {b})" for a, b in pairs]
    return "[" + ", ".join(items) + "]"

print(fmt(zip_lists([1, 2, 3], [4, 5, 6])))
print(fmt(zip_lists([1, 2], [3])))
print(fmt(zip_lists([], [1])))

File: benchmark/comparison/results/blind_ts/17.ts
function zip(a: number[], b: number[]): string[] {
  const len = Math.min(a.length, b.length);
  const result: string[] = [];
  for (let i = 0; i < len; i++) {
    result.push(`Pair(${a[i]}, ${b[i]})`);
  }
  return result;
}

function formatList(lst: string[]): string {
  return "[" + lst.join(", ") + "]";
}

console.log(formatList(zip([1, 2, 3], [4, 5, 6])));
console.log(formatList(zip([1, 2], [3])));
console.log(formatList(zip([], [1])));
